crates/pathfinder/src/consensus/inner/p2p_task.rs

```diff
@@ -409,18 +440,39 @@ fn commit_finalized_block(storage: Storage, finalized_block: FinalizedBlock) ->
     Ok(())
 }
 
+struct ConnectionWrapper<'a>(Transaction<'a>);
+
+// Declaring the transaction thread-safe obviously doesn't _make_ it
+// thread-safe - it still must not be used across await points, and
+// moreover it must be checked manually that it isn't used across
+// await points, because the compiler doesn't see that.
+unsafe impl Send for ConnectionWrapper<'_> {}
+
 async fn handle_incoming_proposal_part(
```

Comment:
Maybe instead of introducing a pseudo-Send connection wrapper this fn should just become blocking in its entirety and it's called then should be wrapped in `spawn_blocking`? Most of the time it's doing heavy lifting anyway - right (DB+execution)?